pipelines/mix/mix.py

```diff
@@ -118,176 +112,53 @@ def generate_result_plots(self):
-                title = f'{str(city)}-{str(daytype)}'
-                day_df = city_df.loc[city_df.daytype == daytype]
-                day_df = day_df.append(day_df.iloc[0])
-                day_df = day_df.reset_index()
-                plot = day_df[plotting_components].plot(kind='area', title=title, grid=True, xticks=ticks, ylim=(0, 1), linewidth=2, color=['green','yellow','brown','blue','grey','black','red'])
-                plt.xlabel('Hour-of-Day')
-                plt.ylabel('Load (pu. total peak)')   
-                fig = plot.get_figure()
-                image_index_based_name = '{0:0=2d}'.format(image_index)
-                fig.savefig(f'{commercial_mix_hour_norm_plots_dir}/{image_index_based_name}.png')
-                plt.close(fig)
-                image_index += 1   
+        self.mix_type_plotting(mix_type=commercial_mix_hour_norm, mix_type_name='URBAN', normalization='hour_norm', directory=commercial_mix_hour_norm_plots_dir)
 
         logger.info('GENERATING hour_norm MIXED MIX PLOTS')
-
-        image_index = 0
-        for idx, city in enumerate(mixed_mix_hour_norm.target.unique()):
-            city_df = mixed_mix_hour_norm.loc[mixed_mix_hour_norm.target == city]
-            max_total = city_df[plotting_components].sum(axis=1).max()
-            max_val = 1 if max_total <= 1 else int(max_total) + 1
-            for ydx, daytype in enumerate(city_df.daytype.unique()):
-                title = f'{str(city)}-{str(daytype)}'
-                day_df = city_df.loc[city_df.daytype == daytype]
-                day_df = day_df.append(day_df.iloc[0])
-                day_df = day_df.reset_index()
-                plot = day_df[plotting_components].plot(kind='area', title=title, grid=True, xticks=ticks, ylim=(0, 1), linewidth=2, color=['green','yellow','brown','blue','grey','black','red'])
-                plt.xlabel('Hour-of-Day')
-                plt.ylabel('Load (pu. total peak)')   
-                fig = plot.get_figure()
-                image_index_based_name = '{0:0=2d}'.format(image_index)
-                fig.savefig(f'{mixed_mix_hour_norm_plots_dir}/{image_index_based_name}.png')
-                plt.close(fig)
-                image_index += 1   
+        self.mix_type_plotting(mix_type=mixed_mix_hour_norm, mix_type_name='MIXED', normalization='hour_norm', directory=mixed_mix_hour_norm_plots_dir)
 
         logger.info('GENERATING hour_norm RURAL MIX PLOTS')
+        self.mix_type_plotting(mix_type=rural_mix_hour_norm, mix_type_name='RURAL', normalization='hour_norm', directory=rural_mix_hour_norm_plots_dir)
 
-        image_index = 0
-        for idx, city in enumerate(rural_mix_hour_norm.target.unique()):
-            city_df = rural_mix_hour_norm.loc[rural_mix_hour_norm.target == city]
+    def mix_type_plotting(self, mix_type, mix_type_name, normalization, directory):
+        ######## Plotting helper function
+        import numpy as np
+        import matplotlib.pyplot as plt
```

Comment:
Same as the other comment, move these to the top of the file

Reply:
fixed